kernel/boot_info/src/uefi.rs

```diff
@@ -0,0 +1,241 @@
+use crate::ElfSectionFlags;
+use bootloader_api::info;
+use core::{
+    iter::{Iterator, Peekable},
+    ops::Range,
+};
+use kernel_config::memory::{KERNEL_OFFSET, KERNEL_STACK_SIZE_IN_PAGES, PAGE_SIZE};
+use memory_structs::{PhysicalAddress, VirtualAddress};
+
+// TODO: Ideally this would be defined in nano_core.
```

Comment:
explain why we can't do it that way 
(cuz of cyclic deps, right? the nano_core must depend on boot_info)